src/Orleans/Logging/LoggerImpl.cs

```diff
@@ -0,0 +1,614 @@
+        /// <param name="args">Any arguments to the format string.</param>
+        public override void Info(int logCode, string format, params object[] args)
+        {
+            Log(logCode, Severity.Info, format, args, null);
+        }
+        /// <summary>
+        /// Writes a log entry at the Warning severity level, with the specified log id code.
+        /// Warning is suitable for problem conditions that the system or application can handle by itself,
+        /// but that the administrator should be aware of.
+        /// Typically these are situations that are expected but that may eventually require an administrative
+        /// response if they recur.
+        /// Warning is lower than Error.
+        /// </summary>
+        /// <param name="logCode">The log code associated with this message.</param>
+        /// <param name="format">A standard format string, suitable for String.Format.</param>
+        /// <param name="args">Any arguments to the format string.</param>
+        public override void Warn(int logCode, string format, params object[] args)
+        {
+            Log(logCode, Severity.Warning, format, args, null);
+        }
+        /// <summary>
+        /// Writes a log entry at the Warning severity level, with the specified log id code.
+        /// Warning is suitable for problem conditions that the system or application can handle by itself,
+        /// but that the administrator should be aware of.
+        /// Typically these are situations that are expected but that may eventually require an administrative
+        /// response if they recur.
+        /// Warning is lower than Error.
+        /// </summary>
+        /// <param name="logCode">The log code associated with this message.</param>
+        /// <param name="message">The warning message to log.</param>
+        /// <param name="exception">An exception related to the warning, if any.</param>
+        public override void Warn(int logCode, string message, Exception exception = null)
+        {
+            Log(logCode, Severity.Warning, message, new object[] { }, exception);
+        }
+        /// <summary>
+        /// Writes a log entry at the Error severity level, with the specified log id code.
+        /// Error is suitable for problem conditions that require immediate administrative response.
+        /// </summary>
+        /// <param name="logCode">The log code associated with this message.</param>
+        /// <param name="message">The error message to log.</param>
+        /// <param name="exception">An exception related to the error, if any.</param>
+        public override void Error(int logCode, string message, Exception exception = null)
+        {
+            Log(logCode, Severity.Error, message, new object[] { }, exception);
+        }
+
+        #endregion
+
+        #region Internal log methods using ErrorCode categorization.
+
+        internal void Verbose(ErrorCode errorCode, string format, params object[] args)
```

Comment:
I think these methods should either be moved to the base Logger class or even better become internal extension methods for the Logger class. Users should not have to reference this internal concrete class just to be able to use this syntactic sugar.